core/lexer.mll

```diff
@@ -134,6 +134,7 @@ let keywords = [
  "var"      , VAR;
  "where"    , WHERE;
  "with"     , WITH;
+ "match"    , MATCH;
```

Comment:
I had suggested `fun (...) match {case ...}` but I think @dhil preferred to use `switch` and not add a new keyword, which I'd also be happy with.

Reply:
On the other hand, since `switch` is paired with `case` and doesn't handle curried arguments, maybe having a different keyword is justifiable - it just seems strange to have both, with different case syntax.